### DISCUSSION TITLE
Welcome to zha-device-handlers Discussions!

### DISCUSSION THREAD

**Discussion by dmulcahey** [zigpy/zha-device-handlers#625] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nemesis24** [comment]
I will be the first, "it is a small step for man, but a giant step for community" !!!

Hello, I'm french, my English is really poor, so excuse me !! (i try to not use google translate)
I'm using zha with a lot of xiaomi products, and ikea plugs.
I'm very proud about all the things you've done in this project... Thank you very much @dmulcahey and the others contributors.

For me it's really difficult to understand how i can make a quirk, i'm really curious about it.

For my ideas

- could edit in line a quirk or some quirks like automations ( a quirk editor )

- could define the type of the device ( example  lumi.sensor_switch.aq2 could be switch, light, fan, cover)

-  An "interlock software" mode

I hope one of my products (lumi.lock v1) could have a support soon.

If i can help the community it would be great !

**Comment by MattWestb** [comment]
I will **NOT** be the first :-((((
But I was the second one and definitely **NOT** the last one writing somthing here :-))

Then i was not able getting the LIDL Zigbee Christmas tree lights so i was buying 2  LIDL _TZ3000_1obwwnmq  3x plug with USB. Its looks like its working OK and I can steering the 3 outlets without problems in HA.

![LIDLUSB](https://user-images.githubusercontent.com/49618193/101636141-2ee9a180-3a2b-11eb-8ebc-a18f356811c2.PNG)

I was seeing that deCONZ was patching the endpoints then only the first one is having basic cluster. https://github.com/dresden-elektronik/deconz-rest-plugin/pull/3838#issue-534630185

Is it necessarily / needed doing the same with one quirk for getting the function working ZHA / HA and not getting problems in the future ?

```
{
  "node_descriptor": "NodeDescriptor(byte1=1, byte2=64, mac_capability_flags=142, manufacturer_code=4098, maximum_buffer_size=82, maximum_incoming_transfer_size=82, server_mask=11264, maximum_outgoing_transfer_size=82, descriptor_capability_field=0)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0x010a",
      "in_clusters": [
        "0x0000",
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006"
      ],
      "out_clusters": [
        "0x000a",
        "0x0019"
      ]
    },
    "2": {
      "profile_id": 260,
      "device_type": "0x010a",
      "in_clusters": [
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006"
      ],
      "out_clusters": []
    },
    "3": {
      "profile_id": 260,
      "device_type": "0x010a",
      "in_clusters": [
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006"
      ],
      "out_clusters": []
    },
    "242": {
      "profile_id": 41440,
      "device_type": "0x0061",
      "in_clusters": [],
      "out_clusters": [
        "0x0021"
      ]
    }
  },
  "manufacturer": "_TZ3000_1obwwnmq",
  "model": "TS011F",
  "class": "zigpy.device.Device"
}
```

And thanks to our devs and all community members that is making all this things great in our systems !!!

And for christmas i wishing all of us in the community one great **Home Smart Home** christmas !!

**Comment by Adminiuga** [comment]
ZHA really doesn't care if there's a single basic cluster on one of the endpoints or if there are multiple basic clusters. It always would use the 1st basic cluster it founds when processing all of the endpoints.

**Comment by lexcao1729** [comment]
我就想知道这玩意儿怎么用……

**Comment by yatsik77** [comment]
Can anybody help with custom quirk debug?
[https://community.home-assistant.io/t/custom-quirk-for-diyruz-airsense/374314](url)

**Reply by Hedda** [reply]
> Can anybody help with custom quirk debug? [https://community.home-assistant.io/t/custom-quirk-for-diyruz-airsense/374314](url)

@yatsik77 FYI, quirk requests should instead be reported as individual "Device support request" issue for zha-device-handlers, see:

https://github.com/zigpy/zha-device-handlers/issues/new/choose

But before doing so be sure to search existing issues there as zha-quirks issues with this device might already have been reported:

https://github.com/zigpy/zha-device-handlers/issues

PS: The documentation for the ZHA integration tries to explain what is supported by default and why a quirk might be needed:

https://www.home-assistant.io/integrations/zha#knowing-which-devices-are-supported

https://www.home-assistant.io/integrations/zha#zha-exception-and-deviation-handling

https://www.home-assistant.io/integrations/zha#reporting-issues

https://www.home-assistant.io/integrations/zha#debug-logging
